actionpack/lib/action_dispatch/routing/mapper.rb

```diff
@@ -637,6 +637,12 @@ def mount(app, options = nil)
             options = app
             app, path = options.find { |k, _| k.respond_to?(:call) }
             options.delete(app) if app
+
+            raise ArgumentError, "A rack application must be specified" unless app.respond_to?(:call)
+            ActionDispatch.deprecator.warn(<<-MSG.squish)
+              Mapping an engine route with a hash key name is deprecated and
```

Comment:
Maybe?
```suggestion
              Mounting an engine route with a hash key name is deprecated and
```

Reply:
Yeah, I think that's easier to understand language. I'll update to "Mounting" and "Drawing"